duckduckgo/iOS
# Sync limit exceeded

Task/Issue URL: https://app.asana.com/0/0/1205727656906843/f

**Description**: When the user exceeds the bookmarks or credentials limit when syncing an error is shown in the sync menu. Pixel are fired for exceeding the item number limit and request size limit .https://www.figma.com/file/mpFLwzGJFlbsmyD1JkAwt2/%F0%9F%94%84-Sync-6?type=design&node-id=4126-82029&mode=design&t=EH5eKCX7rbZMsIVb-0



**Steps to test this PR**:
1. Go to the sync menu and check no error is reported
2. Set up a proxy tool and intercept responses from https://dev-sync-use.duckduckgo.com/sync/bookmarks
3. Exit and reenter the sync menu intercept the response and change the response code to 409
4. Check that you the following pixel has been fired: **_m_d_sync_bookmarks_count_limit_exceeded_daily_**
5. Check on the sync menu the error appears as illustrated in figma.
6. Check that clicking on managing bookmarks takes you to the bookmarks manager.
7. Add a new bookmark
8. Check that the error disappeared from the sync menu
9.  Set up a proxy tool and intercept responses from https://dev-sync-use.duckduckgo.com/sync/bookmarks
10.  Exit and reenter the sync menu intercept the response and change the response code to 413
11. Check that you the following pixel has been fired: **_m_d_sync_bookmarks_request_size_limit_exceeded_daily_**
12. Check on the sync menu the error appears as illustrated in figma.
13. Check that clicking on managing bookmarks takes you to the bookmarks manager.
14. Repeat the same steps for credentials (responses from https://dev-sync-use.duckduckgo.com/sync/credentials should be intercepted,pixels should be  m_d_sync_credentials_count_limit_exceeded_daily and m_d_sync_credentials_request_size_limit_exceeded_daily)

<!—
Before submitting a PR, please ensure you have tested the combinations you expect the reviewer to test, then delete configurations you *know* do not need explicit testing.

Using a simulator where a physical device is unavailable is acceptable.
—>

**Copy Testing**:


**Orientation Testing**:


**Device Testing**:


**OS Testing**:


**Theme Testing**:


—
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
